crd/multitenancy/api/v1alpha1/nodeinfo.go

```diff
@@ -45,8 +45,8 @@ type NodeInfoStatus struct {
 }
 
 type DeviceInfo struct {
-	DeviceType  DeviceType `json:"deviceType,omitempty"`
-	DeviceCount int        `json:"deviceCount"`
+	DeviceType     DeviceType `json:"deviceType,omitempty"`
+	MacAddressList []string   `json:"macAddressList"`
```

Comment:
please document these.
do we expect a device to have multiple MACs?

Reply:
mac address is being used as a unique identifier here, so image there are 3 nics with device type "acn.azure.com/vnet-nic", those three macs would show in this list

Reply:
why wouldn't there be 3 `DeviceInfos` in the status instead? It is also a slice
```go
// NodeInfoStatus defines the observed state of NodeInfo
type NodeInfoStatus struct {
	// +kubebuilder:validation:Optional
	DeviceInfos []DeviceInfo `json:"deviceInfos,omitempty"`
}
```

Reply:
Suppose you wanted to represent 4 devices, two IB and two vnet, it would look like this:
```json
{
  "deviceInfos": [
    {
      "deviceType": "acn.azure.com/vnet-nic",
      "macAddressList": [
        "ab:cd:ef:12:34:56",
        "12:34:56:78:9a:bc"
      ]
    },
    {
      "deviceType": "acn.azure.com/infiniband-nic",
      "macAddressList": [
        "11:22:33:44:55:66",
        "22:33:44:55:66:77"
      ]
    }
}
```

Reply:
However, we should re-think this now. Originally, this structure was intended only to convey counts, e.g. "2 IB and 2 vnet NICS". Now that we're passing device "IDs" anyway, perhaps this should be something like
```go
type DeviceType string // actually this is already declared in podnetwork.go I think

type DeviceInfo {
    DeviceType DeviceType
    MACAddress string
}

type NodeInfoStatus struct {
    DeviceInfos []DeviceInfo
}
```

Reply:
I was going say map[type][]uid but slice of individual items with type/uid field is probably easier to reason about

Reply:
@thatmattlong Hmm.. yeah this looks good and more intuitive to understand. Let me update
```
type DeviceInfo {
    DeviceType DeviceType
    MACAddress string
}

type NodeInfoStatus struct {
    DeviceInfos []DeviceInfo
}
```